### PR TITLE
CORDA-499: Remove further `Kt` classes

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by andr3ej** [comment]
Why are we doing this?

**Comment by rnicoll** [comment]
Many of the external developers at banks are using Corda from Java, and the intent is to make the interop cleaner by removing Kotlin-specific parts, or designs that expose we're using Kotlin.

**Comment by andr3ej** [comment]
So this isn't necessary for internal stuff?

**Comment by rnicoll** [comment]
It's not necessary if the changes to code under `internal` packages would make the API less usable. Are you thinking the `JarInputStream.extractFile()` change? It seemed out of place where it was, but   I'm not absolutely committed to that part.

**Comment by andr3ej** [comment]
My concern is that Kt classes are going to sneak back in almost immediately (they're a useful automagic Kotlin feature and I've not been told not to use them) so I don't see the value of these changes in places unseen by users.
Specifically:
* extractFile isn't API and should stay internal, ArtemisUtils should probably join it in internal
* Surely we are sparing users from the AMQP stuff

and one inline comment.

File: core/src/main/kotlin/net/corda/core/utilities/NetworkHostAndPort.kt [hunk 1]
```diff
@@ -2,38 +2,43 @@ package net.corda.core.utilities
...
  */
 @CordaSerializable
 data class NetworkHostAndPort(val host: String, val port: Int) {
     companion object {
-        internal val invalidPortFormat = "Invalid port: %s"
-        internal val unparseableAddressFormat = "Unparseable address: %s"
-        internal val missingPortFormat = "Missing port: %s"
+        internal const val INVALID_PORT_FORMAT = "Invalid port: %s"
+        internal const val UNPARSEABLE_ADDRESS_FORMAT = "Unparseable address: %s"
+        internal const val MISSING_PORT_FORMAT = "Missing port: %s"
```
Comment (andr3ej) [inline core/src/main/kotlin/net/corda/core/utilities/NetworkHostAndPort.kt]: Why should these be uppercase? They are only non-private constants for unit testing purposes.

**Comment by andr3ej** [comment]
Also, just noticed my formatter (which I haven't changed from the default config) adds a blank line after the @ file directive.

File: core/src/main/kotlin/net/corda/core/utilities/NetworkHostAndPort.kt (hunk 1 above)
Comment (rnicoll) [reply core/src/main/kotlin/net/corda/core/utilities/NetworkHostAndPort.kt]: Following the naming convention for constants, i.e. https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html

Comment (andr3ej) [reply core/src/main/kotlin/net/corda/core/utilities/NetworkHostAndPort.kt]: I see that Kotlin says that too https://kotlinlang.org/docs/reference/properties.html#compile-time-constants but it seems to be to self-document things that can be used in annotations. These aren't for annotations so I think it's unnecessarily ugly. @mikehearn ?

**Comment by rnicoll** [comment]
Rebased, reverted the JAR extraction move, and reformatted the two files I added an annotation to.

File: node-api/src/main/kotlin/net/corda/nodeapi/ArtemisUtils.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@file:JvmName("ArtemisUtils")
```
Comment (andr3ej) [inline node-api/src/main/kotlin/net/corda/nodeapi/ArtemisUtils.kt]: Did you run the formatter on this one?

Comment (rnicoll) [reply node-api/src/main/kotlin/net/corda/nodeapi/ArtemisUtils.kt]: Done now

**Pull request merged**